### DISCUSSION TITLE
same source input many sinks problem

### DISCUSSION THREAD

**Discussion by lddlww** [vectordotdev/vector#22063] [Q&A] [answered]
same source： kafka
different many sinks：databend，elasticsearch

simple config.yaml:  difference sinks using same source input ```source-kafka```
```
vim sources/source-kafka.yaml
type: kafka

vim sinks/sink-databend.yaml
type: databend
inputs:
 - source-kafka

vim sinks/sink-es.yaml
type: elasticsearch
inputs:
 - source-kafka
```

if elasticsearch run abnormal and blocked，which will result  databend sink be blocked，it is right? reference follow code：
https://github.com/vectordotdev/vector/blob/029a2ffc33c66fe1d8b755ad309b695317b1f542/src/topology/builder.rs#L255
https://github.com/vectordotdev/vector/blob/029a2ffc33c66fe1d8b755ad309b695317b1f542/lib/vector-core/src/fanout.rs#L427

because``` sender_group.send ``` function will pending, if one of sinks run abnormal?

if not, am i missing something?

best wishes

**Comment by jszwedko** [comment, marked as answer]
> if elasticsearch run abnormal and blocked，which will result databend sink be blocked，it is right? reference follow code：

This is expected. https://vector.dev/docs/about/concepts/#backpressure has more details. You can avoid this by configuring the buffer for a sink to drop events rather than apply back-pressure.
